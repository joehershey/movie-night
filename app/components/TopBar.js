import React from "react";
import {
  StyleSheet,
  View,
  Text,
  Image,
  TouchableWithoutFeedback,
} from "react-native";

import { COLORS, STYLES } from "../assets/saved";
import { FontAwesome5 } from "@expo/vector-icons";

import fetch from "cross-fetch";

function TopBar(props) {
  return (
    <View style={STYLES.topBar}>
      <TouchableWithoutFeedback
        onPress={() => props.navigation.navigate(props.screen)}
      >
        <View
          style={{
            flex: 1,
            alignItems: "center",
            justifyContent: "center",
          }}
        >
          <FontAwesome5 name="arrow-left" color="white" size={20} />
          <Text style={{ color: "white", margin: 5 }}>{props.message}</Text>
        </View>
      </TouchableWithoutFeedback>

      {/* Logo */}
      <View
        style={{
          flex: 2,
          alignItems: "center",
          paddingBottom: 10,
        }}
      >
        <Image
          style={{
            flex: 1,
            width: 100,
            height: 50,
            marginBottom: 0,
          }}
          source={require("../assets/MovieNightLogo2.png")}
        ></Image>
      </View>

      {/* This code checks if the current screen is the landing page or not, displaying 
            a link to the user's profile if yes and blank space if no. */}
      {props.screen == "Welcome" ? (
        <TouchableWithoutFeedback
          onPress={() => props.navigation.navigate("Profile")}
        >
          <View
            style={{
              flex: 1,
              alignItems: "center",
              justifyContent: "center",
            }}
          >
            <FontAwesome5 name="user" color="white" size={20} />
            <Text style={{ color: "white", margin: 5 }}>{"User Profile"}</Text>
          </View>
        </TouchableWithoutFeedback>
      ) : (
        <View style={{ flex: 1 }}></View>
      )}
      {/* End of spacer / user profile*/}
    </View>
  );
}

export default TopBar;
