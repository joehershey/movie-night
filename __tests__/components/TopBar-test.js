import TopBar from "../../app/components/TopBar";
import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { TouchableWithoutFeedback } from "react-native";
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

describe('Test Top Bar', () => {
    beforeEach( async () => {
        global.tree = renderer.create(<TopBar />);
    });

    it("Top bar renders properly", async () => {
        await expect(global.tree).toMatchSnapshot();
    });

    it("Check for onPress navigation function and simulate press", () => {
        navigate = jest.fn();
        props = {
            navigation: {
                navigate
            },
            screen: "test"
        }
        const container = shallow(<TopBar {...props} />);
        const onPress = container.find(TouchableWithoutFeedback).at(0); // finds onPress => props.navigation.navigate(props.screen)
        onPress.props().onPress(); // calls navigate(props.screen)
        expect(onPress.exists());
        expect(navigate.mock.calls[0][0]).toBe("test")
    });

    it("Check for user profile button from Welcome screen and simulate press", () => {
        navigate = jest.fn();
        props = {
            navigation: {
                navigate
            },
            screen: "Welcome"
        }
        const container = shallow(<TopBar {...props} />);
        const onPress = container.find(TouchableWithoutFeedback).at(1); // finds onPress => props.navigation.navigate("Profile")
        onPress.props().onPress(); // calls navigate("Profile")
        expect(onPress.exists());
        expect(navigate.mock.calls[0][0]).toBe("Profile");
    });
});