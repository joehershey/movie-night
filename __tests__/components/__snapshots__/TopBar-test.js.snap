// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Top Bar Top bar renders properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1d1f1e",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "color": "white",
          "margin": 5,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "paddingBottom": 10,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "flex": 1,
          "height": 50,
          "marginBottom": 0,
          "width": 100,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
